odoo/orm/environments.py

```diff
@@ -200,6 +200,21 @@ def cache(self):
         """Return the cache object of the transaction."""
         return self.transaction.cache
 
+    @property
+    def _access_context(self):
+        """Return the context values used by the access caches."""
+        return (self.uid, *self['ir.rule']._compute_domain_context_values())
+
+    @functools.cached_property
+    def _access_cache(self):
+        """Get the cache for READ access (model -> ids -> bool)."""
```

Comment:
TODO model -> tuple[set[id], set[id]] (accessible, inaccessible)

Reply:
This is actually a bad idea.
- in terms of performance on python 3.12, the simple `in` is faster than `get` (~5%), but other operations I am using have similar performance or are slower  -- overall performance is similar in both cases
- in terms of memory footprint on python 3.12, dicts are smaller
  ```
  In [3]: sys.getsizeof({i: True for i in range(100)})
  Out[3]: 4688
  In [4]: sys.getsizeof(set(range(100)))
  Out[4]: 8408
  ```
- in terms of use, when checking for new ids, you have to do operations on both sets, so it's around twice slower - to avoid inconsistencies where a key is in both sets (which does not happen in dict)